### DISCUSSION TITLE
Node-Red-Contrib-Mindconnect Error - Network Error Ocurred Request, reason: getaddrinfo  EAI_AGAIN

### DISCUSSION THREAD

**Discussion by HermannSilva** [mindsphere/node-red-contrib-mindconnect#180] [Q&A] [answered]
I'm currently running the block **node-red-contrib-mindconnect** on **Node-RED version 2.2.2**. 

The block is working, uploading data to Mindsphere, but i got the message **"Network Error Ocurred Request, reason: getaddrinfo  EAI_AGAIN".**

I've checked the details: internet connection, NTP on device, certificates, etc. Evertything seems to be ok.

What could be the reasons and possible solutions to the proble.

**Comment by sn0wcat** [comment, marked as answer]
that is a DNS error - the node relies on the node.js networking stack at it expects the DNS to work.

**Reply by sn0wcat** [reply]
try to run 

 
```shell
nslookup southgate.eu1.mindsphere.io 
```

on the device.

**Comment by HermannSilva** [comment]
Thanks for the fast reply, when i execute the command. I get the following status (DNS Request timed out) :

![image](https://github.com/mindsphere/node-red-contrib-mindconnect/assets/145358657/7e6f89de-84fa-48b9-a2a3-0940e38c51b6)

Do i need to configure a DNS Address for the Mindsphere server? Is there any doc about it?

**Reply by sn0wcat** [reply]
you have to configure DNS in general - either use automatic DNS from your internet provider , specific DNS server, or one of the free available DNS servers on the internet

see e.g. 
https://www.pdq.com/blog/change-dns-server-settings-in-windows/
